cpp/src/arrow/compute/light_array.h

```diff
@@ -348,11 +349,13 @@ class ARROW_EXPORT ExecBatchBuilder {
   ExecBatch Flush();
 
   int num_rows() const { return values_.empty() ? 0 : values_[0].num_rows(); }
+  bool is_full() const { return num_rows() == num_rows_max(); }
 
   static int num_rows_max() { return 1 << kLogNumRows; }
 
  private:
   static constexpr int kLogNumRows = 15;
+  static constexpr int64_t kAlignment = 512;
```

Comment:
How confident are we that `512` will work for all disks?  What will happen if it doesn't?

Reply:
512 is the number I see posted most often on the internet, but supposedly the requirements can vary by file system or something. Anyway, I think in the worst case `O_DIRECT` will be a no-op.